src/TypeDefinition.js

```diff
@@ -207,14 +207,14 @@ export type NavigationUriAction = {
 export type NavigationStackViewConfig = {
   mode?: 'card' | 'modal',
   headerMode?: HeaderMode,
-  headerComponent?: ReactClass<HeaderProps<*>>,
   cardStyle?: Style,
   transitionConfig?: () => TransitionConfig,
   onTransitionStart?: () => void,
   onTransitionEnd?: () => void,
 };
 
 export type NavigationStackScreenOptions = NavigationScreenOptions & {
+  header?: React.Element<*> | ((HeaderProps) => React.Element<*>),
```

Comment:
So, if its an element it would not accept any HeaderProps?

Instead of `((HeaderProps) => React.Element<*>)`, does it make sense to do `ReactClass<HeaderProps>`?

Also I think we should have a `?` here after the `header?:`, to signify that null is valid.